p2p/pkg/rpc/provider/service_test.go

```diff
@@ -36,8 +37,16 @@ func (t *testRegistryContract) RegisterAndStake(opts *bind.TransactOpts, blsPubl
 	return types.NewTransaction(1, common.Address{}, nil, 0, nil, nil), nil
 }
 
+func (t *testRegistryContract) Stake(opts *bind.TransactOpts) (*types.Transaction, error) {
+	t.topup = opts.Value
+	return types.NewTransaction(1, common.Address{}, nil, 0, nil, nil), nil
+}
+
 func (t *testRegistryContract) GetProviderStake(_ *bind.CallOpts, address common.Address) (*big.Int, error) {
-	return t.stake, nil
+	if t.stake.Cmp(big.NewInt(0)) == 0 {
+		return big.NewInt(0), errors.New("no stake")
```

Comment:
Zero value of pointer is `nil` not `big.NewInt(0)`.

Reply:
I am initializing it to big.NewInt(0). Which is why this check. I had to change this anyway as the contract returns 0 even if stake is not registered. So I have used a different contract function now.